### PR TITLE
improve behavior in case of range data loss

### PR DESCRIPTION
**Pull request by baumanta**
Make range data loss safer, before a Hold was sent immediately if range data was missing for more than 0.5s. But the velocity inputs were fed back the same as they came in. Meaning until the mode switch went through, the vehicle would proceed at full speed.

Also switching the mode immediately is not desirable (e.g. in the middle of a takeoff). So we now the limit the xy velocities to zero until we really do not get any data (after 5s) and then switch to hold.

Also the "no range data" warnings were output at full rate, now they are throttled to once a second.

### CODE REVIEW

**Comment by hamishwillee** [comment]
Can you add a section "what happens if data is lost" or similar? http://docs.px4.io/master/en/computer_vision/collision_prevention.html

File: src/lib/collision_prevention/CollisionPrevention.cpp [hunk 1]
```diff
@@ -105,6 +105,18 @@ hrt_abstime CollisionPrevention::getElapsedTime(const hrt_abstime *ptr)
 	return hrt_absolute_time() - *ptr;
 }
 
+bool CollisionPrevention::is_active()
+{
+	bool is_active = _param_cp_dist.get() > 0;
```
Comment (jkflying) [inline src/lib/collision_prevention/CollisionPrevention.cpp]: It seems there's another variable called `is_active` that Jenkins is complaining about. So this should be renamed.

File: src/lib/collision_prevention/CollisionPreventionTest.cpp [hunk 2]
```diff
@@ -325,6 +325,74 @@ TEST_F(CollisionPreventionTest, testPurgeOldData)
...
+
+	// AND: a parameter handle
+	param_t param = param_handle(px4::params::CP_DIST);
+	float value = 10; // try to keep 10m distance
+	param_set(param, &value);
+	cp.paramsChanged();
+
+	// AND: an obstacle message without any obstacle
+	obstacle_distance_s message;
+	memset(&message, 0xDEAD, sizeof(message));
```
Comment (julianoes) [inline src/lib/collision_prevention/CollisionPreventionTest.cpp]: Huh? I don't get that.

Comment (baumanta) [reply src/lib/collision_prevention/CollisionPreventionTest.cpp]: To be honest, I don't know either... I copied the skeleton from another test and did not question that line. I tried running the test without that line and it still passes. Should I remove the memsets from all the test, if the pass without them?

Comment (jkflying) [reply src/lib/collision_prevention/CollisionPreventionTest.cpp]: I did that in the early tests when I was making sure that the messages were actually getting sent/received correctly. Not sure if it is as critical any more.

**Pull request merged**